### PR TITLE
Implement Convolution 2D

### PR DESCRIPTION
**Pull request by daemyung**
## 🙏 Describe the pull request

Implement Convolution 2D but it isn't optimized.

## ✅ Checklist

- [x] Code follows the project's coding conventions and style.
- [x] Tests have been added or updated to cover the changes.
- [x] Documentation has been updated, if necessary.

### CODE REVIEW

File: trident/kernel/conv2d.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                channel_block_size: triton.language.constexpr, weight_block_size: triton.language.constexpr):
+        program_id = triton.language.program_id(0)
+
+        in_batch = language.batch(program_id, out_channels, out_height, out_width)
+        out_channel = language.channel(program_id, out_channels, out_height, out_width)
+        out_height = language.row(program_id, out_height, out_width)
+        out_width = language.col(program_id, out_width)
+
+        channel_block = triton.language.arange(0, channel_block_size)
+        channel_mask = channel_block < in_channels
```
Comment (kakao-steve-ai) [inline trident/kernel/conv2d.py:35]: 제가 아직 코드를 파편적으로 이해하고 있어서 질문드립니다. 상위 caller(oeration/conv2d.py)에서 channel_block_size를 triton.next_power_of_2(in_channels)으로 넘겼는데, 그러면 여기 비교 구문이 항상 True 일거 같아서요. 이 mask 의 용도가 무엇인가요?

Comment (kakao-steve-ai) [reply trident/kernel/conv2d.py:35]: 아 질문이 잘못되었네요. in_channels 이 10 이었다고 하면, channel_block_size 는 16 이 되고, 10 ~ 15은 False 가 되는군요. 앞에 코멘트는 무시하셔도 됩니다.

Comment (daemyung) [reply trident/kernel/conv2d.py:35]: @rbbang 네 이해하신게 맞습니다. Triton이 GPU를 효율적으로 사용하기 위해 2의 승수를 요구하는 경우가 있습니다.

File: trident/language/language.py [hunk 2]
```diff
@@ -50,6 +50,11 @@ def row(index, num_rows, num_cols):
     return (index % (num_rows * num_cols)) // num_cols
 
 
+@triton.jit
+def sum3d(x):
+    return triton.language.sum(triton.language.ravel(x), 0)
```
Comment (mejai1206) [inline trident/language/language.py:55]: 꼭 3d가 아니고 Nd여도 잘 동작할 것 같은데 맞을까요?

Comment (daemyung) [reply trident/language/language.py:55]: @mejai1206 네 맞습니다. `sum`으로 변경해둘께요.

**Review by kakao-steve-ai — APPROVED** [review]
구현 좋아보입니다! 한가지 궁금한점은, 정의를 찾아보니 torch.nn.functional.conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1)와 같이 되어 있는데, 다양한 stride, padding, dilation 값에 대한 구현은 TODO로 남겨놓은 것이 맞는지요?

**Comment by daemyung** [comment]
> 구현 좋아보입니다! 한가지 궁금한점은, 정의를 찾아보니 torch.nn.functional.conv2d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1)와 같이 되어 있는데, 다양한 stride, padding, dilation 값에 대한 구현은 TODO로 남겨놓은 것이 맞는지요?

@rbbang 네 맞습니다. 빠진 기능은 추후에 필요할때 구현할 계획입니다.
